src/fetcher.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from "fs";
+import fetch from "node-fetch";
+import { JSDOM } from "jsdom";
+
+fetchIDLs();
+
+interface IDLSource {
+    url: string;
+    title: string;
+}
+
+async function fetchIDLs() {
+    const file = fs.readFileSync(`${__dirname}/../inputfiles/idlSources.json`, { encoding: "utf-8" });
+    const idlSources = JSON.parse(file) as IDLSource[];
+    for (const source of idlSources) {
+        const idl = await fetchIDL(source);
+        fs.writeFileSync(`${__dirname}/../inputfiles/idl/${source.title}.widl`, idl + '\n');
+    }
+}
+
+async function fetchIDL(source: IDLSource) {
+    const response = await fetch(source.url);
+    const dom = new JSDOM(await response.text());
+    const elements = Array.from(dom.window.document.querySelectorAll("pre.idl:not(.extract),code.idl-code"));
```

Comment:
totally random, and not related to this change; it would be super cool if in the future we can extract the description from the widl file and put it as a comment on the declaration..